torch/ao/sparsity/experimental/data_sparsifier/base_data_sparsifier.py

```diff
@@ -38,39 +56,198 @@ def __init__(self, data_list: Optional[List[Tuple[str, Any]]] = None, **defaults
     def prepare(self):
         raise NotImplementedError("this function is undefined for this class")
 
+    def _extract_weight(self, data):
+        if isinstance(data, torch.Tensor):
+            return data
+
     def add_data(self, name: str, data, **config):
         r""" Configures and parametrizes the internal container model with name and data
 
         Note: The container model is private to the BaseDataSparsifier class
         """
-        pass
+        assert type(data) in SUPPORTED_TYPES, \
+            f'data type unsupported for {name}'
+
+        if name in self.state:
+            # If the named data already exists - replace
+            warnings.warn("Replacing existing data of the same name. - Did you mean a different name?")
+
+        local_args = copy.deepcopy(self.defaults)
+        local_args.update(config)
+        self.data_groups[name] = local_args
+
+        weight = self._extract_weight(data)
+
+        # Bookkeeping in the container class
+        mask = local_args.get('mask', torch.ones_like(weight))
+        param_class = local_args.get('parametrization', utils.FakeSparsity)  # change once public_api for utils is fixed!
+        param = nn.Parameter(weight, requires_grad=False)
+
+        if name in self.state:
+            # If named data exists, check if parametrized
+            if parametrize.is_parametrized(self._container, name):
+                # If parametrized, squash mask
+                self.squash_mask(names=[name], leave_parametrized=False)
+            self._container.get_parameter(name).data = data  # overwrite the data
+        else:
+            setattr(self._container, name, param)
+        parametrize.register_parametrization(self._container, name, param_class(mask))
+        self.state[name]['mask'] = mask
+        return getattr(self._container, name)
+
+    def get_data(self, name: str, return_sparsified: bool = True):
+        r"""Returns weight tensor (or data)
+        Args:
+            - name: name of the data to be returned
+            - return_sparsified: returns weight tensor after applying parametrization if True
+                else - returns the original version (non-parametrized)
+        """
+        if name not in self.data_groups:
+            raise ValueError("data with specified name does not exist")
 
-    def get_data(self, name: str):
-        pass
+        if not return_sparsified:
+            if not parametrize.is_parametrized(self._container, name):
+                raise ValueError("mask squashed - original mask value does not exist")
```

Comment:
nit: If it's not parametrized, and the assumption is that the squash_mask was already called, shouldn't this just return the tensor from the container? If the intention is to return the parametrized vs. original, I think a better name for the argument shoud be something like

```
def get_data(self, name: str, return_original: bool = True):
```

Reply:
makes sense.